.github/workflows/tests-docker.yml

```diff
@@ -47,10 +47,13 @@ jobs:
 
       - name: Run tests
         run: |
-          nox --session tests-${{ matrix.python }} -- --real-caikit
+          nox --session tests-${{ matrix.python }} -- --real-caikit --cov-report=xml
 
       - name: Upload coverage report
         uses: codecov/codecov-action@v3.1.4
         if: always()
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        with:
+          files: coverage.xml
+          fail_ci_if_error: true
```

Comment:
Turns out we were getting low coverage because the reports for the docker tests were not being uploaded :facepalm: